jaxrs-clients/src/test/java/feign/TextDelegateDecoderTest.java

```diff
@@ -78,6 +79,13 @@ public void testUsesStringDecoderWithTextPlain() throws Exception {
     }
 
     @Test
+    public void testCannotReturnStringWithMediaTypeJson() throws Exception {
+        assertThatThrownBy(() -> service.getJsonString("foo"))
```

Comment:
the server is returning the raw string in this example, that behavior is the same.